tests/draft2020-12/unevaluatedProperties.json

```diff
@@ -809,5 +809,223 @@
+        },
+        "tests": [
+            {
+                "description": "no extra properties",
+                "data": {
+                    "foo": {
+                        "bar": "test"
+                    }
+                },
+                "valid": true
+            },
+            {
+                "description": "uncle keyword evaluation is not significant",
+                "data": {
+                    "foo": {
+                        "bar": "test",
+                        "faz": "test"
+                    }
+                },
+                "valid": false
+            }
+        ]
+    },
+    {
+        "description": "property is evaluated in a nephew schema to unevaluatedProperties",
+        "schema": {
+            "type": "object",
+            "properties": {
+                "bar": {
+                    "type": "string"
+                }
+            },
+            "unevaluatedProperties": false,
+            "anyOf": [
+                {
+                    "properties": {
+                        "faz": {
+                            "type": "string"
+                        }
+                    }
+                }
+            ]
+        },
+        "tests": [
+            {
+                "description": "no extra properties",
+                "data": {
+                    "bar": "test"
+                },
+                "valid": true
+            },
+            {
+                "description": "all in-place applicators are evaluated before unevaluatedProperties",
+                "data": {
+                    "bar": "test",
+                    "faz": "test"
+                },
+                "valid": true
+            }
+        ]
+    },
+    {
+        "description": "in-place applicator siblings, allOf has unevaluated",
+        "schema": {
+            "type": "object",
+            "allOf": [
+                {
+                    "properties": {
+                        "foo": true
+                    },
+                    "unevaluatedProperties": false
+                }
+            ],
+            "anyOf": [
+                {
+                    "properties": {
+                        "bar": true
+                    }
+                }
+            ]
+        },
+        "tests": [
+            {
+                "description": "base case: both properties present",
+                "data": {
+                    "foo": 1,
+                    "bar": 1
+                },
+                "valid": false
+            },
+            {
+                "description": "in place applicator siblings, bar is missing",
+                "data": {
+                    "foo": 1
+                },
+                "valid": true
+            },
+            {
+                "description": "in place applicator siblings, foo is missing",
+                "data": {
+                    "bar": 1
+                },
+                "valid": false
+            }
+        ]
+    },
```

Comment:
I'm not sure this test is necessary. It's effectively the same test as "property is evaluated in an uncle schema to unevaluatedProperties". It just uses a different kind of applicator (`allOf` instead of `properties`). I'd prefer to keep this one because it's simpler. Also, I believe it could be simplified and still test the same thing.

```suggestion
    {
        "description": "in-place applicator siblings, allOf has unevaluated",
        "schema": {
            "type": "object",
            "allOf": [
                {
                    "properties": {
                        "foo": true
                    },
                    "unevaluatedProperties": false
                },
                {
                    "properties": {
                        "bar": true
                    }
                }
            ]
        },
        "tests": [
            {
                "description": "base case: both properties present",
                "data": {
                    "foo": 1,
                    "bar": 1
                },
                "valid": false
            },
            {
                "description": "in place applicator siblings, bar is missing",
                "data": {
                    "foo": 1
                },
                "valid": true
            },
            {
                "description": "in place applicator siblings, foo is missing",
                "data": {
                    "bar": 1
                },
                "valid": false
            }
        ]
    },
```

Reply:
But this would be basically the same as "cousin unevaluatedProperties, true and false, false with properties" on line 780. The evaluated and unevaluated properties have the same parent keyword - although structurally not directly under the parent. I think the idea with "in-place applicator siblings, allOf has unevaluated" is testing that an entirely different keyword branch does not affect unevaluatedProperties.

Reply:
I agree that this is the only test needed, but I don't think it should be simplified.

Reply:
I don't think it's redundant with the other test at all -- because unevaluatedProperties could be special-casing the properties keyword and the schemas underneath it, but actually it needs to consider the results of *all* in-place applicators. (Also the cousin tests passed in my implementation, whereas the new anyOf tests did not, because of the order in which keywords are evaluated and results collected. The allOf tests passed, only because allOf is always evaluated before anyOf -- so switching the order of the subschemas caused the results to change, which is also incorrect.)

Reply:
If it exposed a bug in your implementation that was not caught by any other test, definitely keep it.